lib/tests/2wp.js

```diff
@@ -29,6 +32,87 @@ const setupBridgeDonation = async (rskTxHelpers, btcTxHelper) => {
   await donateToBridge(rskTxHelpers[0], btcTxHelper, donatingBtcAddressInformation, DONATION_AMOUNT);
 };
 
+const addInputs = (tx, sendersUtxosInfo) => {
+  sendersUtxosInfo.flatMap(senderUtxosInfo => senderUtxosInfo.utxos).forEach(uxto => {
+    tx.addInput(Buffer.from(uxto.txid, 'hex').reverse(), uxto.vout);
+  });
+};
+
+const addChangeOutputs = (tx, sendersInfo, sendersChange) => {
+  sendersChange.forEach((change, index) => {
+    if(change > 0) {
+      tx.addOutput(
+        bitcoinJsLib.address.toOutputScript(sendersInfo[index].btcSenderAddressInfo.address, btcTxHelper.btcConfig.network),
+        Number(btcToSatoshis(change))
+      );
+    }
+  });
+};
+
+const addOutputsToFed = (tx, outputsToFed) => {
+  outputsToFed.forEach(outputAmount => {
+    tx.addOutput(
+      bitcoinJsLib.address.toOutputScript(federationAddress, btcTxHelper.btcConfig.network),
+      Number(btcToSatoshis(outputAmount))
+    );
+  });
+};
+
+const pushPegin = async (btcPeginTxHash, expectedUtxosCount) => {
+  await waitForBitcoinTxToBeInMempool(btcTxHelper, btcPeginTxHash);
+  await mineForPeginRegistration(rskTxHelper, btcTxHelper);
+  await ensurePeginIsRegistered(rskTxHelper, btcPeginTxHash, expectedUtxosCount);
+};
+
+const getSendersInfo = async (initialBtcSenderBalancesInBtc) => {
+  return await Promise.all(initialBtcSenderBalancesInBtc.map(initialBtcSenderBalance => createSenderRecipientInfo(rskTxHelper, btcTxHelper, 'legacy', initialBtcSenderBalance + btcTxHelper.getFee())));
+};
+
+const getSendersBtcAddressBalances = async (sendersInfo) => {
+  return await Promise.all(sendersInfo.map(senderInfo => btcTxHelper.getAddressBalance(senderInfo.btcSenderAddressInfo.address)));
+};
+
+const getSendersUtxosInfo = async (sendersInfo, btcSenderAmountsToSendToFed) => {
+  return await Promise.all(sendersInfo.map((senderInfo, index) => btcTxHelper.selectSpendableUTXOsFromAddress(senderInfo.btcSenderAddressInfo.address, btcSenderAmountsToSendToFed[index])));
+};
+
+const getSendersChange = (sendersUtxosInfo) => {
+  return sendersUtxosInfo.map(senderUtxosInfo => senderUtxosInfo.change - btcTxHelper.getFee());
+};
+
+const getSendersPrivateKeys = (sendersInfo) => {
+  return sendersInfo.map(senderInfo => senderInfo.btcSenderAddressInfo.privateKey);
+};
+
+const getTotalAmountToFed = (outputsToFed) => {
+  return outputsToFed.reduce((total, amount) => total + amount, 0);
+};
+
+const getFinalRskRecipientBalances = async (sendersInfo) => {
+  return await Promise.all(sendersInfo.map(senderInfo => rskTxHelper.getBalance(senderInfo.rskRecipientRskAddressInfo.address)));
+};
+
+const legacyPeginTestsHappyPath = [
```

Comment:
These are not simple legacy pegins. I consider these legacy pegins as border cases, as well as uncommon/complex legacy pegins, at least those ones sending multiple outputs to the fed, since there's no reason for wallet to do that.

That being said, I would add a prefix to this variable like `uncommon/complex-LegacyPeginTests` . wdyt?